pkg/database/connection.go

```diff
@@ -50,12 +52,21 @@ func initializePool() pgxpoolmock.PgxPool {
 		klog.Fatal("Error parsing database connection configuration. ", configErr)
 	}
 
-	conn, err := pgxpool.ConnectConfig(context.Background(), config)
-	if err != nil {
-		klog.Error("Unable to connect to database: %+v\n", err)
-		// TODO: We need to retry the connection until successful.
-	} else {
-		klog.Info("Successfully connected to database!")
+	retry := 0
+	var conn *pgxpool.Pool
+	var err error
+	for {
+		conn, err = pgxpool.ConnectConfig(context.Background(), config)
+		if err != nil {
+			klog.Errorf("Unable to connect to database: %+v\n", err)
+			waitMS := int(math.Min(float64(retry*15*1000), float64(cfg.MaxBackoffMS)))
```

Comment:
This database connection is critical, so I want to reduce the wait between retries and max backoff.
How about we add 500ms each retry up to 30000ms (30 seconds) max backoff?

Reply:
Done